redisplus/json/commands.py

```diff
@@ -137,6 +135,21 @@ def jsonset(self, name, path, obj, nx=False, xx=False):
             pieces.append("XX")
         return self.execute_command("JSON.SET", *pieces)
 
+    def jsonset(self, name, path, obj, nx=False, xx=False, decode_keys=False):
+        """
+        Set the JSON value at key ``name`` under the ``path`` to ``obj``.
+
+        ``nx`` if set to True, set ``value`` only if it does not exist.
+        ``xx`` if set to True, set ``value`` only if it exists.
+        ``decode_keys`` If set to True, the keys of ``obj`` will be decoded with utf-8.
+        """
+        if decode_keys:
```

Comment:
Let's build the command first, and return if decode_keys is false (i.e have the return first.

Then the rest is outside of the if.